apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,268 @@
+/*Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <sys/types.h>
+#include <task_manager/task_manager.h>
+#include "tc_common.h"
+
+#define TM_SAMPLE_NAME "itc_tm_sample"
+#define TM_NOPERM_NAME "tm_noperm"
+#define TM_SAMPLE_MSG  "test_msg"
+#define TM_INVALID_HANDLE      -2
+#define TM_UNREGISTERED_HANDLE 99
+#define TM_INVALID_TIMEOUT     -2
+#define TM_INVALID_PERMISSION  -2
+#define TM_INVALID_UPPER_RANGE_PERMISSION 3
+#define TM_INVALID_UPPER_RANGE_HANDLE 33
+
+static int tm_sample_handle;
+static int tm_noperm_handle;
+static bool flag;
+static task_info_t *sample_info;
+static task_info_t *second_sample_info;
+
+static void test_handler(void *info)
+{
+	flag = !strncmp(info, TM_SAMPLE_MSG, strlen(TM_SAMPLE_MSG));
+}
+
+int itc_sample_main(int argc, char *argv[])
+{
+	int ret;
+
+	tm_noperm_handle = task_manager_register(TM_NOPERM_NAME, TM_TASK_PERMISSION_DEDICATE, 100);
+
+	ret = task_manager_set_unicast_cb(test_handler);
+	if (ret != OK) {
+		printf("ERROR : fail to set handler\n");
+	} while (1) {
+                usleep(1);
+        }       // This will be dead at utc_task_manager_terminate_p
+        return ret;
+
+}
+
+/**
+ * @testcase         itc_task_manager_clean_info_twice
+ * @brief            Get task_manager_getinfo_with_handle twice
+ * @scenario         Get task_manager_getinfo_with_handle twice
+ * @apicovered       task_manager_getinfo_with_handle
+ * @precondition     NA
+ * @postcondition    NA
+ */
+static void itc_task_manager_clean_info_twice(void)
```

Comment:
Please add cleaninfo for second_sample_info.
If not, there will be memory leaks.

Reply:
added and code also commited